src/spartan_router.erl

```diff
@@ -13,30 +13,24 @@
 -spec(upstreams_from_questions(dns:questions()) -> ordsets:ordset(upstream())).
 upstreams_from_questions([#dns_query{name=Name}]) ->
     Labels = spartan_app:parse_upstream_name(Name),
-    lists:map(fun normalize_ip/1, find_upstream(Name, Labels));
```

Comment:
Ah, yeah parsing user input is fine, I just meant that the function that parses user input should validate and convert it to the proper form. Then the aggregator function (which was previously normalize_ips) doesn't have to re-parse everything again.